src/options.cpp

```diff
@@ -90,13 +90,18 @@ options parse_options(int argc, char** argv) {
       next_path_for_config_search = arg_value;
     } else if (const char* arg_value =
                    parser.match_option_with_value("--vim-file-bufnr"sv)) {
+      o.has_vim_file_bufnr = true;
       int bufnr;
       from_chars_result result =
           from_chars(&arg_value[0], &arg_value[std::strlen(arg_value)], bufnr);
       if (*result.ptr != '\0' || result.ec != std::errc{}) {
         o.error_unrecognized_options.emplace_back(arg_value);
         continue;
       }
+      if (next_vim_file_bufnr != std::nullopt) {
+        o.warning_vim_bufnr_without_file.emplace_back(
+            next_vim_file_bufnr.value());
```

Comment:
Minor: I think we should store the original value, making `warning_vim_bufnr_without_file` a `vector<const char*>` instead. That way we don't need to convert the `int` to a string when printing the warning.

The main problem with my idea is that `next_vim_file_bufnr` only stores the `int`, but this should be easy to fix.